shadowaead/stream.go

```diff
@@ -40,38 +42,58 @@ func (w *writer) Write(b []byte) (int, error) {
 // writes to the embedded io.Writer. Returns number of bytes read from r and
 // any error encountered.
 func (w *writer) ReadFrom(r io.Reader) (n int64, err error) {
-	for {
-		buf := w.buf
-		payloadBuf := buf[2+w.Overhead() : 2+w.Overhead()+payloadSizeMask]
-		nr, er := r.Read(payloadBuf)
-
-		if nr > 0 {
-			n += int64(nr)
-			buf = buf[:2+w.Overhead()+nr+w.Overhead()]
-			payloadBuf = payloadBuf[:nr]
-			buf[0], buf[1] = byte(nr>>8), byte(nr) // big-endian payload size
+	readAndEnctypt := func(buf []byte) (n int, err error) {
+		payloadBuf := buf[2+w.Overhead():]
```

Comment:
this may results a panic due to index out of range(L49).

max(n) known as payloadBuf size in L47 = bufLength - 2 - overhead, so to get bytes in this line will be `2 + overhead + buf length - 2 - overhead + overhead = overhead + buf length` has a extra overhead.

so here should be `payloadBuf := buf[2+w.Overhead() : 2+w.Overhead()+payloadSizeMask]`

Reply:
Thank you for the feedback, I fixed this issue in the latest commit.